azext_iot/operations/hub.py

```diff
@@ -2434,6 +2434,96 @@ def iot_hub_distributed_tracing_update(
     )
 
 
+def iot_hub_connection_string_show(
+    cmd,
+    hub_name=None,
+    resource_group_name=None,
+    policy_name='iothubowner',
+    key_type=KeyType.primary.value,
+    show_all=False,
+    include_eventhub=False
+):
+    discovery = IotHubDiscovery(cmd)
+
+    if hub_name is None:
+        hubs = discovery.get_iothubs(resource_group_name)
+        if hubs is None:
+            raise CLIError("No IoT Hub found.")
+
+        def conn_str_getter(hub):
+            return _get_hub_connection_string(
+                discovery,
+                hub,
+                policy_name,
+                key_type,
+                show_all,
+                include_eventhub
+            )
+        return [{'name': hub.name, 'connectionString': conn_str_getter(hub)} for hub in hubs]
+    hub = discovery.find_iothub(hub_name, resource_group_name)
+    if hub:
+        conn_str = _get_hub_connection_string(
+            discovery,
+            hub,
+            policy_name,
+            key_type,
+            show_all,
+            include_eventhub
+        )
+        return {'connectionString': conn_str if show_all else conn_str[0]}
+
+
+def _get_hub_connection_string(
+    discovery,
+    hub,
+    policy_name,
+    key_type,
+    show_all,
+    include_eventhub
+):
+
+    policies = []
+    if show_all:
+        policies.extend(
+            discovery.get_policies(
+                hub.name,
+                hub.additional_properties['resourcegroup']
+            )
+        )
+    else:
+        policies.append(
+            discovery.find_policy(
+                hub.name,
+                hub.additional_properties['resourcegroup'],
+                policy_name
+            )
+        )
+
+    if include_eventhub:
+        cs_template_eventhub = 'Endpoint={};SharedAccessKeyName={};SharedAccessKey={};EntityPath={}'
+        endpoint = hub.properties.event_hub_endpoints['events'].endpoint
+        entityPath = hub.properties.event_hub_endpoints['events'].path
+        return [
+            cs_template_eventhub.format(
+                endpoint,
+                p.key_name,
+                p.secondary_key if key_type == KeyType.secondary.value else p.primary_key,
+                entityPath
+            ) for p in policies if "serviceconnect" in p.rights.value.lower()
```

Comment:
We should state this filtering behavior in the help description for the command so users aren't confused why a policy connection string doesn't show up when --include-eventhub (to be --default-eventhub)  is passed in.

Reply:
Mentioned in one of the examples in help content.